File: LeetCode/missing-number.cpp
// https://leetcode.com/problems/missing-number/

class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int sum = 0;
        
        for(int i=1;i<=nums.size();i++) {
            sum += i;
            sum -= nums[i-1];
        }
        
        return sum;
    }
};

File: LeetCode/missing-number.java
class Solution {
    // we use bit operators
    // first XOR the input array
    // since we have distinct natural nos. from 1 to n in the array with one missing number
    // if we XOR the result from the first loop
    // then we cancel out all duplicates
    // and get the missing number
    public int missingNumber(int[] nums) {
        int x = nums[0];
        for(int i=1;i<nums.length;i++) {
            x = x ^ nums[i];
        }
        
        for(int i=1;i<=nums.length;i++) {
            x = x ^ i;
        }
        
        return x;
    }
}